src/wgsl_reflect.js

```diff
@@ -379,12 +379,26 @@ export class WgslReflect {
             }
         }
 
-        const info = WgslReflect.TypeInfo[type.name];
-        if (info) {
-            return {
-                align: Math.max(explicitAlign, info.align),
-                size: Math.max(explicitSize, info.size)
-            };
+        {
+            const info = WgslReflect.TypeInfo[type.name];
+            if (info) {
+                const divisor = type.format === 'f16' ? 2 : 1
+                return {
+                    align: Math.max(explicitAlign, info.align / divisor),
+                    size: Math.max(explicitSize, info.size / divisor)
+                };
+            }
+        }
+
+        {
+            const info = WgslReflect.TypeInfo[type.name.substring(0, type.name.length - 1)]

```

Comment:
I would add explicit TypeInfo keys for f and h variants of vec and mat, which would eliminate the need for this block.

Reply:
I started to do that but size is important. There are are not just f and h there's also i and u. `vec4i`, `vec4u`, `vec4f`, `vec4h`.  Putting all the types in the table will be 4x-8x larger

Reply:
We can leave it as the string manip version for now.

Sorry for the delayed reply, got stuck on another project.